src/texture/tex_impl.rs

```diff
@@ -47,8 +48,8 @@ pub struct TextureImplementation {
 
 impl TextureImplementation {
     /// Builds a new texture.
-    pub fn new<P>(display: &Display, format: TextureFormatRequest,
-                  data: Option<(ClientFormat, Vec<P>)>, width: u32, height: Option<u32>,
+    pub fn new<'a, P: Clone + 'a>(display: &Display, format: TextureFormatRequest,
```

Comment:
Could you add ` + Clone + 'a` in the `where` clause at the end of the function, since there's already one?

And while you're at it, realign the parameters vertically.